### PR TITLE
Support stagic page generation in app router

### PR DESCRIPTION
**Pull request by alebedev**
## Checklist before requesting a review

- [ ] I have performed a self-review of my code

### CODE REVIEW

**Comment by alebedev** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/4185?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#4185** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/4185?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#4181** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/4181?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @alebedev and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: apps/store/src/app/[locale]/appDebug/[[...slug]]/ProductCmsPage.tsx [hunk 1]
```diff
@@ -49,19 +50,21 @@ export const ProductCmsPage = async ({ locale, story }: ProductCmsPageProps) =>
...
-                <ProductPageBlock blok={story.content} />
-                <PageDebugDialog />
-                <ProductPageViewTracker />
-                <PageBannerTriggers blok={story.content} />
-                <BankIdDialog />
-              </BankIdContextProvider>
-            </ProductReviewsMetadataProvider>
-          </ProductPageTrackingProvider>
-        </PriceIntentContextProvider>
+        <Suspense>
```
Comment (alebedev) [inline apps/store/src/app/[locale]/appDebug/[[...slug]]/ProductCmsPage.tsx:53]: Getting rid of this Suspense requires refactoring PriceIntentContextProvider and getting rid of useSearchParams. Planned as a part of app router migration

File: apps/store/src/app/[locale]/appDebug/[[...slug]]/page.tsx [hunk 2]
```diff
@@ -76,14 +78,35 @@ export async function generateMetadata({ params }: Props): Promise<Metadata> {
   return result
 }
 
+export async function generateStaticParams({
+  params,
+}: {
+  params: Pick<CmsPageRoutingParams, 'locale'>
+}): Promise<Array<{ slug: Array<string> }>> {
+  // TODO: Remove when main CMS route is removed from pages router
+  if (process.env.APP_ROUTER_GENERATE_CMS_PAGES !== 'true') {
```
Comment (alebedev) [inline apps/store/src/app/[locale]/appDebug/[[...slug]]/page.tsx:87]: Generating top pages is relatively quick, but we can still save 15-20s by not doing it eagerly

File: apps/store/src/services/storyblok/storyblok.serverOnly.ts [hunk 3]
```diff
@@ -1,62 +1,116 @@
...
-  await primeCache()
-  const { data } = await storyblokApi.getStory(`${locale}/${slug}`, {
-    version,
-    resolve_links: 'url',
-    resolve_relations: `reusableBlockReference.reference,${BLOG_ARTICLE_CONTENT_TYPE}.categories,page.abTestOrigin`,
-  })
-  return data.story as T
-}
+// GOTCHA: All arguments must be primitives for react.cache to work
+export const getStoryBySlug = cache(
```
Comment (alebedev) [inline apps/store/src/services/storyblok/storyblok.serverOnly.ts:24]: Not sure react cache is beneficial on top of fetch cache, reminder to myself to measure the effect

Comment (alebedev) [reply apps/store/src/services/storyblok/storyblok.serverOnly.ts:24]: Surprisingly, removing React cache makes dev mode ~2x slower - all request are cached still according to the logs. It has something to do with deduplication of work between renderer and metadata generator.  Moved caching to `page.txs` to make it more explicit and keep storyblok API more consistent

**Review by guilhermespopolin — APPROVED** [review]
Looks good
Rafael has mentioned something about `getStoryblokApiWithCache` [here](https://hedviginsurance.slack.com/archives/C041SD67G82/p1714756428524449?thread_ts=1714721216.085789&cid=C041SD67G82). Not sure if we can do something about it in this PR.

**Pull request merged**